apps/opik-backend/src/main/java/com/comet/opik/domain/TraceService.java

```diff
@@ -167,23 +171,33 @@ public Mono<Long> create(TraceBatch batch) {
                 .distinct()
                 .toList();
 
-        return Mono.deferContextual(ctx -> {
-            String workspaceId = ctx.get(RequestContext.WORKSPACE_ID);
-            String userName = ctx.get(RequestContext.USER_NAME);
+        // Delete existing attachments for all traces in the batch before processing
+        // This prevents duplicate attachments when the SDK sends the same trace data multiple times
+        Set<UUID> traceIds = dedupedTraces.stream()
+                .map(Trace::id)
+                .collect(Collectors.toSet());
 
-            Mono<List<Trace>> resolveProjects = Flux.fromIterable(projectNames)
-                    .flatMap(projectService::getOrCreate)
-                    .collectList()
-                    .map(projects -> bindTraceToProjectAndId(dedupedTraces, projects))
-                    .flatMapMany(Flux::fromIterable)
-                    .flatMap(trace -> attachmentStripperService.stripAttachmentsFromTrace(trace, workspaceId, userName,
-                            trace.projectName()))
-                    .collectList();
-
-            return resolveProjects
-                    .flatMap(traces -> template.nonTransaction(connection -> dao.batchInsert(traces, connection))
-                            .doOnSuccess(__ -> eventBus.post(new TracesCreated(traces, workspaceId, userName))));
-        });
+        return attachmentService.deleteByEntityIds(EntityType.TRACE, traceIds)
+                .then(Mono.deferContextual(ctx -> {
+                    String workspaceId = ctx.get(RequestContext.WORKSPACE_ID);
+                    String userName = ctx.get(RequestContext.USER_NAME);
+
+                    Mono<List<Trace>> resolveProjects = Flux.fromIterable(projectNames)
+                            .flatMap(projectService::getOrCreate)
+                            .collectList()
+                            .map(projects -> bindTraceToProjectAndId(dedupedTraces, projects))
+                            .flatMapMany(Flux::fromIterable)
+                            .flatMap(trace -> attachmentStripperService.stripAttachmentsFromTrace(trace, workspaceId,
+                                    userName,
```

Comment:
For future implementation, could we save the hash of the attachment and compare it when processing it? That way, we would avoid uploading and deleting attachments when the input may not have changed.